src/RamProvenanceExistenceCheck.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2019, The Souffle Developers. All rights reserved
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file RamProvenanceExistenceCheck.cpp
+ *
+ * Implementation of RAM Provenance Existence Check Analysis
+ *
+ ***********************************************************************/
+
+#include "RamProvenanceExistenceCheck.h"
+#include "RamVisitor.h"
```

Comment:
I'm not seeing where RamVisitor.h is needed. RamTypes is used inside getKey, though.
add
#include "RamOperation.h"
#include "RamTypes.h"
remove 
#include "RamVisitor.h"